server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -550,11 +562,15 @@ private void reroute(RoutingAllocation allocation) {
         assert AutoExpandReplicas.getAutoExpandReplicaChanges(allocation.metadata(), allocation).isEmpty()
             : "auto-expand replicas out of sync with number of nodes in the cluster";
         assert assertInitialized();
-
+        long rerouteStartTimeNS = System.nanoTime();
         removeDelayMarkers(allocation);
 
         allocateExistingUnassignedShards(allocation);  // try to allocate existing shard copies first
         shardsAllocator.allocate(allocation);
+        ClusterManagerMetrics.recordLatency(
```

Comment:
Can we move the latency computation to the `ClusterManagerMetrics` class wherever possible ? I am thinking something along the lines of
```
Metric metric = CM.startOperation(CM.rerouteHistogram)
<the actual method call>
metric.endOperation() //records the latency
```

Reply:
As discussed, due to no strict checks present on the way units are defined while declaring metrics, abstracting out the value part would be a bit difficult. Will give more thoughts to this, while we add annotations.